pgxn/neon/libpagestore.c

```diff
@@ -431,6 +434,11 @@ pageserver_connect(shardno_t shard_no, int elevel)
 	}
 	PG_END_TRY();
 
+	if (broke_from_loop)
```

Comment:
should this happen before L429 to keep the same control flow semantics as before?